app/gui/src/presenter.rs

```diff
@@ -125,11 +138,37 @@ impl Model {
                 } else {
                     warn!("Project creation failed: no ProjectManagingAPI available.");
                 }
-            })
+            }))
         } else if let Some(template) = template {
             error!("Invalid project template name: {template}");
         };
     }
+
+    /// Open a project by name or ID. If no project with the given name exists, it will be created.
+    #[profile(Task)]
+    fn open_or_create_project(&self, project: ProjectToOpen) {
+        let controller = self.controller.clone_ref();
+        crate::executor::global::spawn(with_progress_indicator(|| async move {
+            if let Ok(managing_api) = controller.manage_projects() {
+                if let Err(error) = managing_api.open_or_create_project(project).await {
+                    error!("Cannot open or create project: {error}.");
```

Comment:
Is this error not producing double dot at the end?

Reply:
> Is this error not producing double dot at the end?

As a side note: this is why I think we should not make error messages (in any returned "Err") sentences (starting with capital and with dot at the end), because until they are not printed to the user we don't know if they are the entire sentence, or will become a part of another one.

Reply:
I think there is a simple solution for that. We can always use this instead:

```
error!("Cannot open or create project. {error}");
```

And now, you always have nicely formatted errors and you can always assume they are sentences.

Reply:
Shall I just remove the period for now? Changing colons to periods would make it inconsistent with the rest of the codebase, right?

Reply:
Let's not preserve consistency but rather use `error!("Cannot open or create project. {error}");` in this PR. I little inconsistency in error messages is not a serious issue, and we don't have time to update the whole project.